server/server/src/upload_plugin.rs

```diff
@@ -0,0 +1,73 @@
+use std::fs;
+
+use actix_multipart::form::{tempfile::TempFile, MultipartForm};
+use actix_web::{
+    post,
+    web::{self, Data},
+    HttpResponse,
+};
+
+use base64::{prelude::BASE64_STANDARD, Engine};
+use serde::Deserialize;
+use serde_json::json;
+use service::{
+    backend_plugin::plugin_provider::{
+        BindPluginInput, PluginInstance, PluginType, PluginVariantType,
+    },
+    settings::Settings,
+    static_files::{StaticFileCategory, StaticFileService},
+};
+
+pub fn config_upload_plugin(cfg: &mut web::ServiceConfig) {
+    cfg.service(plugin);
+}
+
+// Should be in plugin_provider
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct UrlParams {
+    r#type: PluginType,
```

Comment:
I assume you're refine this a bit in a later PR?
Or at least add some docs about what these types are?
I think `variant_type` could have a better name but struggling to thing of one right now, maybe `plugin-technology`? Could also just leave that off until we actually have a different type?